[cloudstrapper][agw] Build AGW & Cloudstrapper AWS AMI

quentinDERORY opened the pull request:
## Summary

Automate the generation of AWS AMI for Cloudstrapper and AGW.

## Test Plan

Tested in circle ci
https://app.circleci.com/pipelines/github/magma/magma/22997/workflows/ca34319b-236b-4a6b-9f94-1e1c2937819f/jobs/244737

## Additional Information

quentinDERORY committed:
[cloudstrapper][release] Cloudstrapper & AGW AMI build

Added to the release job running in circleci

Signed-off-by: Quentin, Derory <15911421+quentinDERORY@users.noreply.github.com>

quentinDERORY committed:
[agw][cloudstrapper][ci] Add some comments

Signed-off-by: Quentin, Derory <15911421+quentinDERORY@users.noreply.github.com>

quentinDERORY committed:
[cloudstrapper][agw] Get rid of docker and helm credentials as not needed

Signed-off-by: Quentin, Derory <15911421+quentinDERORY@users.noreply.github.com>

arunuke commented on experimental/cloudstrapper/README.md:
```diff
@@ -368,9 +368,9 @@
 
-  Available tags include: agw,eks,asg,es,rds,efs,natgw,igw,subnet,secgroup,vpc
+  Available tags include: agw,eks,asg,es,rds,efs,natgw,igw,subnet,secgroup,vpc,orc8r,keys
 
```
What does the orc8r tag do?

quentinDERORY replied:
by default the clean playbook was getting rid of orc8r componenets, I just tagged that part to avoid the playbook to fail.

arunuke replied:
This change LGTM. Cleanup inherently needs some more work, specifically for partial deployments, where it exits today if it can't find a certain resource. Ideal behavior should be to add a flag to see if the variable is defined and only then progress. We can connect on this.

quentinDERORY replied:
Now the cleanup playbook does not fails when the resources are not found and just continues.

arunuke commented on experimental/cloudstrapper/playbooks/devops-configure.yaml:
```diff
@@ -5,7 +5,8 @@
     - build-platform
-    - {role: key-manager, tags: keyManager}
+    - {role: key-manager, tags: keyManager, become: true}
+    - clean-authorized-keys
```
clean-authorized-keys appears to clear up the two keys that were created as part of the Cloudstrapper, but it is likely that customers might need this even if they cleanup the current deployment. Would it be good to club it under a "Never" tag so that people don't accidentally clean it up?

arunuke replied:
Just noticed that this clears out the keys inside the devops instance which is perfect. The comment should be applicable to the cleanup of keys in the "cleanup" task.

quentinDERORY replied:
I have added never tag to the key cleanup of the cleanup playbook!

arunuke commented on experimental/cloudstrapper/playbooks/roles/aws-cleanup/tasks/keys.yaml:
```diff
@@ -0,0 +1,11 @@
+- name: Delete KeyHost
+  ec2_key:
+    name: "{{ keyHost }}"
+    state: absent
```
It is possible customers use their pre-existing keys and not the ones that were generated as part of Cloudstrapper. In that case, how can we protect customers from accidentally deleting these keys? If there is a way to ensure that these keys were created as part of CLoudstrapper and not inherited from elsewhere, that would be ideal. Maybe a tag added in keyCreate which can be verified here before deleting?

quentinDERORY replied:
I added tagging of keys and I am checking the tags before deleting them!

quentinDERORY committed:
[ci][cloudstrapper] Add new weekly trigger for the circle ci job

Cleanup playbook does not fail when  the resource does not exists
Added tags to the ec2 keys
Versioning different between the release job and the weekly job

Signed-off-by: Quentin, Derory <15911421+quentinDERORY@users.noreply.github.com>

quentinDERORY committed:
[cloudstrapper][agw] Fix apt_key module issue

Now manually download key before using the module

Signed-off-by: Quentin, Derory <15911421+quentinDERORY@users.noreply.github.com>

quentinDERORY committed:
[cloudstrapper][agw] Fix version suffix

Signed-off-by: Quentin, Derory <15911421+quentinDERORY@users.noreply.github.com>

arunuke commented on experimental/cloudstrapper/playbooks/roles/aws-cleanup/tasks/main.yaml:
```diff
@@ -6,3 +6,8 @@
+  tags: orc8r
+- include: keys.yaml
+  environment:
+    AWS_DEFAULT_REGION: "{{ awsOrc8rRegion }}"
```
I think keyHost and keyBoot are created in the awsAgwRegion and not in the awsOrc8rRegion. Typically, we only deploy Orc8r resources in the awsOrc8rRegion and do practically everything else in the awsAgwRegion.

quentinDERORY replied:
thanks for catching that !

arunuke reviewed (changes requested):
Rest of the changes LGTM except for the awsOrc8rRegion/awsAgwRegion for clearing up keys. Woot !!

quentinDERORY committed:
[cloudstrapper][ci] Fix aws region when deleting keys

The keys are used in the agw region not in the orc8r region!

Signed-off-by: Quentin, Derory <15911421+quentinDERORY@users.noreply.github.com>

arunuke reviewed (approved):
LGTM.

quentinDERORY committed:
[cloudstrapper][agw][ci] Switch region to us-east-1 and add LF parameter

Signed-off-by: Quentin, Derory <15911421+quentinDERORY@users.noreply.github.com>

quentinDERORY committed:
[agw][ami][ci] Update tags for playbooks

Signed-off-by: Quentin, Derory <15911421+quentinDERORY@users.noreply.github.com>
